libportability-gfx/src/impls.rs

```diff
@@ -2878,7 +2879,7 @@ pub extern "C" fn gfxCreateDescriptorPool(
             Ok(pool) => pool,
             Err(oom) => return map_oom(oom),
         },
-        temp_sets: Vec::with_capacity(max_sets),
+        temp_sets: SmallVec::with_capacity(max_sets),
```

Comment:
why are we using `SmallVec` here? It seems to me that descriptor pool is one of the cases that a plain old `Vec` works best

Reply:
The change came from `hal::pso::DescriptorPool::allocate_sets` which takes `&mut SmallVec`. I'm not sure there is a way to keep `Vec` here without changing that interface. I think changing `DescriptorPool::allocate_sets` to take an iterator would be the best approach.

Reply:
ah snap, sorry! We'll be having smallvec here than as you did